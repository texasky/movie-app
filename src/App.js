import './App.css';
import { useState } from 'react';
import MoviesList from './components/MoviesList/MoviesList';

function App() {
  const [movies, setMovies] = useState([
    {
      "Title": "Brian Eno: 1971-1977 - The Man Who Fell to Earth",
      "Year": "2011",
      "imdbID": "tt2296338",
      "Type": "movie",
      "Poster": "https://m.media-amazon.com/images/M/MV5BMTgzNjI4MjY5NV5BMl5BanBnXkFtZTgwOTgwNzA2MDE@._V1_SX300.jpg"
  },
  {
      "Title": "The Who: At Kilburn 1977",
      "Year": "2009",
      "imdbID": "tt1509275",
      "Type": "movie",
      "Poster": "https://m.media-amazon.com/images/M/MV5BMTM4MTM3NmQtZjE0YS00NDNlLTgwYWEtMGUyMmE4YWNjNTgwXkEyXkFqcGdeQXVyMTEzMDkwMTE@._V1_SX300.jpg"
  },
  {
      "Title": "Rainer Werner Fassbinder, 1977",
      "Year": "1977",
      "imdbID": "tt0126297",
      "Type": "movie",
      "Poster": "https://m.media-amazon.com/images/M/MV5BOTk5YmEyYjctN2NmOS00MTlhLWEzOGEtYWE4MjE2ODJhYzJiXkEyXkFqcGdeQXVyMjIzMTk4ODg@._V1_SX300.jpg"
  },
  {
      "Title": "You Weren't There: A History of Chicago Punk 1977 to 1984",
      "Year": "2007",
      "imdbID": "tt1340142",
      "Type": "movie",
      "Poster": "https://m.media-amazon.com/images/M/MV5BMTM0Njk5Njc0Ml5BMl5BanBnXkFtZTcwMjc4NjYxNA@@._V1_SX300.jpg"
  },
  {
      "Title": "Turning Point 1977",
      "Year": "2009",
      "imdbID": "tt1562900",
      "Type": "movie",
      "Poster": "https://m.media-amazon.com/images/M/MV5BNzExNDA5NDMtZDgzOC00ZGNlLTkxYzctNmMxYmIyZGE4OTc3XkEyXkFqcGdeQXVyMjExMzEyNTM@._V1_SX300.jpg"
  },
  {
      "Title": "Rainbow: Live in Munich 1977",
      "Year": "2006",
      "imdbID": "tt1857839",
      "Type": "movie",
      "Poster": "https://m.media-amazon.com/images/M/MV5BOGFhZTE5MGItODhkMi00YTMwLTg1ZDQtNTg0MjZkN2RjYTA2XkEyXkFqcGdeQXVyNzQ5MzY0NjM@._V1_SX300.jpg"
  },
  {
      "Title": "1977",
      "Year": "2007",
      "imdbID": "tt1084943",
      "Type": "movie",
      "Poster": "https://m.media-amazon.com/images/M/MV5BMzllOThiODQtZTI1ZC00ZDJlLWJiOWItMTY5ZjQwNWQ0YWYwXkEyXkFqcGdeQXVyMjQ0MzgwNDc@._V1_SX300.jpg"
  },
  {
      "Title": "No Island: The Palmers Kidnapping of 1977",
      "Year": "2006",
      "imdbID": "tt0819369",
      "Type": "movie",
      "Poster": "https://m.media-amazon.com/images/M/MV5BMTk1OTY2MzExN15BMl5BanBnXkFtZTcwMDIwMjc3OA@@._V1_SX300.jpg"
  },
  {
      "Title": "Eagles: Live at the Capital Centre (March 1977)",
      "Year": "2013",
      "imdbID": "tt11585296",
      "Type": "movie",
      "Poster": "https://m.media-amazon.com/images/M/MV5BYmJlNTU4ZTgtZGVjNC00NzZhLWIyZWUtYmFjZGM1YWQ5MzU3XkEyXkFqcGdeQXVyMTA5NzgwNzM1._V1_SX300.jpg"
  },
  {
      "Title": "Utopia: A Retrospective 1977-1984",
      "Year": "1985",
      "imdbID": "tt0304879",
      "Type": "movie",
      "Poster": "N/A"
  }
  ])

  return (
    <div className="App">
      <MoviesList movies={movies}/>
    </div>
  );
}

export default App;
