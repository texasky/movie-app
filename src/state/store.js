import { applyMiddleware, createStore } from "redux"
import configureStore from 'redux-mock-store'
import reducers from "./reducers/index"
import thunk from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension";

const store = createStore(reducers, composeWithDevTools(applyMiddleware(thunk)));

export default store; 