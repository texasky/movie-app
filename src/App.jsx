import './App.css';
import { useEffect } from 'react';
import MoviesList from './components/MoviesList/MoviesList';
import { useDispatch, useSelector } from 'react-redux';
import { fetchMovies } from './state/action-creator/index'
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import MovieHeader from './components/MovieHeader/MovieHeader';
import MoviePreview from './components/MoviePreview/MoviePreview';
import EmptyComponent from './components/EmptyComponent/EmptyComponent';


const App = () => {
  const dispatch = useDispatch();

  const movies = useSelector(state => state.moviesStore.movies);
  const searchQuery = useSelector(state => state.moviesStore.searchQuery);
  const isMovieOpened = useSelector(state => state.moviesStore.isMovieOpened)

  let emptyComponentText;

  if (!movies && !searchQuery) {
    emptyComponentText = 'Type to search a movie'
  } else if (!movies && searchQuery.length < 3) {
    emptyComponentText = 'Enter at least 3 letters to search'
  } else if (!movies && searchQuery) {
    emptyComponentText = 'Sorry, nothing found'
  }

  useEffect(() => {
    dispatch(fetchMovies('star wars'));
  }, [dispatch])

  useEffect(() => {
    if(searchQuery && !isMovieOpened) dispatch(fetchMovies(searchQuery));
  }, [dispatch, searchQuery, isMovieOpened])

  return (
    <Router>
      <div className="App">
        <MovieHeader/>
        {emptyComponentText && !isMovieOpened && <EmptyComponent text={emptyComponentText}/>}
        <Routes>
          {movies && !isMovieOpened && <Route path='/' element={<MoviesList movies={movies}/>}/>}
          <Route path='/moviepreview/:id' element={<MoviePreview/>}/>
        </Routes>
        {/* {isMovieOpened && <MoviePreview/>} */}
        {/* {(movies && !isMovieOpened) && <MoviesList movies={movies}/>} */}
      </div>
    </Router>
  );
}

export default App;
