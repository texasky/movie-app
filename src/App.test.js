import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';
import { Provider } from 'react-redux';
import store from './state/store'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'

describe('App component tests', () => {
  const mockStore = configureMockStore([thunk])

  test('MovieHeader component render', () => {
    render(<Provider store={store}><App/></Provider>);

    const movieRenderElement = screen.getByTestId('movie-header')

    expect(movieRenderElement).toBeInTheDocument();
  });

  describe('Empty component text', () => {
    test('Type to search a movie', () => {
      const store = mockStore({
        moviesStore: {
          searchQuery: '',
          isMovieOpened: false,
          openedMovieId: '',
        }
      })

      render(<Provider store={store}><App/></Provider>);
      const linkElement = screen.queryByTestId('empty-component');
      expect(linkElement).toBeInTheDocument();
      expect(linkElement).toHaveTextContent('Type to search a movie');
    })

    test('Sorry, nothing found', () => {
      const store = mockStore({
        moviesStore: {
          searchQuery: 'abc',
          isMovieOpened: false,
          openedMovieId: '',
        }
      })

      render(<Provider store={store}><App/></Provider>);
      const linkElement = screen.queryByTestId('empty-component');
      expect(linkElement).toBeInTheDocument();
      expect(linkElement).toHaveTextContent('Sorry, nothing found');
    })

    test('Enter at least 3 letters to search', () => {
      const store = mockStore({
        moviesStore: {
          searchQuery: 'ab',
          isMovieOpened: false,
          openedMovieId: '',
        }
      })

      render(<Provider store={store}><App/></Provider>);

      const linkElement = screen.queryByTestId('empty-component');
      expect(linkElement).toBeInTheDocument();
      expect(linkElement).toHaveTextContent('Enter at least 3 letters to search');
    })

    test('No empty text should be rendered', () => {
      const store = mockStore({
        moviesStore: {
          movies: [{
            Poster: "https://m.media-amazon.com/images/M/MV5BMGVmMWNiMDktYjQ0Mi00MWIxLTk0N2UtN2ZlYTdkN2IzNDNlXkEyXkFqcGdeQXVyODE5NzE3OTE@._V1_SX300.jpg",
            Title: "Harry Potter and the Deathly Hallows: Part 2",
            Type: "movie",
            Year: "2011",
            imdbID: "tt1201607"
          }],
          searchQuery: '',
          isMovieOpened: false,
          openedMovieId: '',
        }
      })

      render(<Provider store={store}><App/></Provider>);

      const linkElement = screen.queryByTestId('empty-component');
      const movieListElement = screen.queryByTestId('movie-list');

      expect(linkElement).not.toBeInTheDocument();
      expect(movieListElement).toBeInTheDocument();
    })
  })

  test('App component matches snapshot', () => {
    const tree = renderer.create(<Provider store={store}><App/></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});