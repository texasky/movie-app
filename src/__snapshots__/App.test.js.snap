// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component tests App component matches snapshot 1`] = `
<div
  className="App"
>
  <div
    className="movie-header"
    data-testid="movie-header"
  >
    <div
      className="movie-header-wrapper"
    >
      <div
        className="movie-app-logo"
        data-testid="movie-app-logo"
      >
        <img
          alt="MOVIE APP"
          height={40}
          src="movie-app-icon.jpeg"
          width={40}
        />
        <span>
          MOVIE APP. TYPE AND SEARCH :)
        </span>
      </div>
      <div
        className="movie-search-bar"
      >
        <input
          aria-label="search-input"
          data-testid="movie-search-input"
          disabled={false}
          id="movie-search-input"
          onChange={[Function]}
          placeholder="Type to search"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="movies-list movies-list-wrapper"
    data-testid="movie-list"
  />
</div>
`;
